src/Admin/AbstractAdmin.php

```diff
@@ -2391,8 +2528,35 @@ public function getFilterTheme()
         return $this->filterTheme;
     }
 
-    public function addExtension(AdminExtensionInterface $extension)
+    public function addExtension(OldAdminExtensionInterface $extension)
     {
+        @trigger_error(
+            'Method "addExtension" is deprecated. Check "addAdminExtension".',
+            E_USER_DEPRECATED
+        );
+
+        $this->extensions[] = $extension;
+    }
+
+    /**
+     * NEXT_MAJOR: remove whole if and add only NewAdminExtensionInterface typehint.
+     *
+     * @param NewAdminExtensionInterface|OldAdminExtensionInterface $extension
+     */
+    public function addAdminExtension($extension)
```

Comment:
Why allow old extensions here?

Reply:
So we can call this method from compiler pass for both old and new extensions

Reply:
I can probably check instance of the extension and just call old or new method instead?

Reply:
Nop, it is a reference and AFAIK you can only do that check on a definition, right?